test/test_nn.py

```diff
@@ -10502,6 +10502,22 @@ def test_log_softmax_cpu(self, dtype=torch.bfloat16):
         self.assertEqual(input.grad.dtype, dtype)
         self.assertEqual(input.grad, inputf.grad.to(dtype), atol=0.1, rtol=0)
 
+    def test_non_last_dim_softmax_cpu(self, dtype=torch.bfloat16):
```

Comment:
Do we have similar tests for other dtypes that could be parameterized and extended by bfloat16 or do they not exist (then we should probably write them since this is also supported for other dtypes)?

Reply:
Thanks for the suggestion. Find the `test_softmax_bfloat16` test case which is only for CUDA before. So we can share and extend this test case.